src/shark/utilityconsole/commands/rkz/FindWeapons.java

```diff
@@ -0,0 +1,505 @@
+                                                          ParameterCriterion.CriteriaQuantity criteriaQuantity) {
+        ParameterCriterion retVal = null;
+
+        // now make a context-sensitive ParameterCriterion based on the data we have parsed from the input data.console command
+        // Depending on the criteria, we will have to instantiate different data containers
+        ParameterCriterion.CriteriaParameter.CriteriaData data = null;
+
+        switch (criteria) {
+            case WEAPON:
+                data = new ParameterCriterion.CriteriaParameter.WeaponCriteriaData(weaponType);
+                break;
+            case WEAPON_WITH_SIZE:
+                data = new ParameterCriterion.CriteriaParameter.WeaponAndSizeCriteriaData(weaponType, weaponSize);
+                break;
+            case WEAPON_PARAMETER:
+                data = new ParameterCriterion.CriteriaParameter.WeaponParameterCriteriaData(weaponParameter);
+                break;
+
+            default:
+                Console.showMessage("Unsupported criteria encountered in constructParameterCriteria! Received " + criteria);
+        }
+        ParameterCriterion.CriteriaParameter cp = new ParameterCriterion.CriteriaParameter(criteria, data);
+        retVal = new ParameterCriterion(cp, criteriaQuantity);
+
+        return retVal;
+    }
+
+    public boolean containsShipSizes(String fullInputString) {
+        boolean retVal = false;
+        retVal = fullInputString.toLowerCase().contains("fighter")
+                || fullInputString.toLowerCase().contains("frigate")
+                || fullInputString.toLowerCase().contains("destroyer")
+                || fullInputString.toLowerCase().contains("cruiser")
+                || fullInputString.toLowerCase().contains("capital") || fullInputString.toLowerCase().contains("capital-ship") || fullInputString.toLowerCase().contains("capital_ship");
+
+        return retVal;
+    }
+
+    public String replaceShipSizesWithOrdinals(String fullInputString) {
+        // We don't really care about the position of the shipsize word,
+        // just replace all of them in one go, it'll hit what it hits, and call it done.
+        return fullInputString.toLowerCase()
+                .replaceAll("fighter", String.valueOf(ShipAPI.HullSize.FIGHTER.ordinal()))
+                .replaceAll("frigate", String.valueOf(ShipAPI.HullSize.FRIGATE.ordinal()))
+                .replaceAll("destroyer", String.valueOf(ShipAPI.HullSize.DESTROYER.ordinal()))
+                .replaceAll("cruiser", String.valueOf(ShipAPI.HullSize.CRUISER.ordinal()))
+                .replaceAll("capital", String.valueOf(ShipAPI.HullSize.CAPITAL_SHIP.ordinal()))
+                .replaceAll("capital-ship", String.valueOf(ShipAPI.HullSize.CAPITAL_SHIP.ordinal()))
+                .replaceAll("capital_ship", String.valueOf(ShipAPI.HullSize.CAPITAL_SHIP.ordinal()));
+    }
```

Comment:
i wont even bother coming up with a better comment than this at the moment.

does this get replaced with `energy/missile/ballistic` instead though, that's the question that makes more sense.

Reply:
done. resolving.